doc/sphinx/language/core/basic.rst

```diff
@@ -55,11 +46,34 @@ graphically using the following kind of blocks:
    Alternatives (either :n:`item__1` or :n:`item__2` or ...).
 
 `Precedence levels
-<https://en.wikipedia.org/wiki/Order_of_operations>`_, when they
-exist, are represented by appending a number to the name of the
-nonterminal (as in :n:`@term100` or :n:`@ltac_expr3`).  Precedence
-levels which are unused by the base grammar are not represented, even
-though they could still be populated by notations or plugins.
+<https://en.wikipedia.org/wiki/Order_of_operations>`_ that are
+implemented in the |Coq| parser are shown in the documentation by
+appending the level to the name of the nonterminal (as in
+:n:`@term100` or :n:`@ltac_expr3`).
+
+.. note::
+
+   |Coq| uses an extensible parser.  Plugins and the :ref:`notation
+   system <syntax-extensions-and-notation-scopes>` can extend the
+   syntax at run time.  Some notations are defined in the prelude,
+   which is loaded by default.  The documented grammar doesn't include
+   these notations.  Precedence levels not used by the base grammar
+   are omitted from the documentation, even though they could still be
+   populated by notations or plugins.
+
+   Furthermore, some parsing rules are only activated in certain
+   contexts (:ref:`interactive proof mode <proofhandling>`,
+   :ref:`custom entries <custom-entries>`...).
+
+.. warning::
+
+   Given the complexity of these parsing rules, it would be extremely
+   difficult to create an external program that can properly parse a
+   |Coq| document.  Therefore, tool writers are advised to delegate
+   parsing to |Coq|, by communicating with it, for instance through
+   `SerAPI <https://github.com/ejgallego/coq-serapi>`_.
```

Comment:
The goal is precisely to be discouraging. We have well-supported solutions, and we want to have people stop attempting creating parsers on their own unless they know exactly what they are doing, or unless they don't care about robustness.

Reply:
OK.  There's a history of people trying to write their own parsers and a) wasting a bunch of their own time because they don't understand the subtleties and/or b) and this wastes the time of other developers?

Reply:
Yes